TexLab LaTeX language server

<!--
IMPORTANT: we are not accepting issues about Termux
installations running on Android 5.x-6.x.
-->

**Package description**

**TexLab**
A cross-platform implementation of the Language Server Protocol providing rich cross-editing support for the LaTeX typesetting system.

Can be used via extensions with NeoVim, Emacs/Spacemacs and others...

**Link to home page and sources**
1. Home page: https://texlab.netlify.com/
2. Source code: https://github.com/latex-lsp/texlab

**Additional information**
Compiling with Rust ran into the following error --->

~~~
 19   master  ✔  cargo build --release                               :~/t/texlab
   Compiling libc v0.2.66
   Compiling autocfg v0.1.7
   Compiling proc-macro2 v1.0.7
   Compiling unicode-xid v0.2.0
   Compiling cfg-if v0.1.10
   Compiling syn v1.0.13
   Compiling lazy_static v1.4.0
   Compiling memchr v2.2.1
   Compiling semver-parser v0.7.0
   Compiling rand_core v0.4.2
   Compiling serde v1.0.104
   Compiling log v0.4.8
   Compiling byteorder v1.3.2
   Compiling ryu v1.0.2
   Compiling siphasher v0.2.3
   Compiling version_check v0.1.5
   Compiling itoa v0.4.4
   Compiling getrandom v0.1.13
   Compiling string_cache_shared v0.3.0
   Compiling ppv-lite86 v0.2.6
   Compiling unicode-xid v0.1.0
   Compiling maybe-uninit v2.0.0
   Compiling either v1.5.3
   Compiling proc-macro2 v0.4.30
   Compiling new_debug_unreachable v1.0.3
   Compiling syn v0.15.44
   Compiling slab v0.4.2
   Compiling precomputed-hash v0.1.1
   Compiling arrayvec v0.4.12
   Compiling fnv v1.0.6
   Compiling futures-core v0.3.1
   Compiling scopeguard v1.0.0
   Compiling bitflags v1.2.1
   Compiling nodrop v0.1.14
   Compiling static_assertions v0.3.4
   Compiling proc-macro-nested v0.1.3
   Compiling futures-sink v0.3.1
   Compiling regex-syntax v0.6.12
   Compiling mac v0.1.1
   Compiling futures-io v0.3.1
   Compiling futures-task v0.3.1
   Compiling arc-swap v0.4.4
   Compiling utf-8 v0.7.5
   Compiling pin-utils v0.1.0-alpha.4
   Compiling pin-project-lite v0.1.1
   Compiling stable_deref_trait v1.1.1
   Compiling bytes v0.5.3
   Compiling matches v0.1.8
   Compiling typenum v1.11.2
   Compiling unicode-segmentation v1.6.0
   Compiling smallvec v1.0.0
   Compiling once_cell v1.2.0
   Compiling siphasher v0.3.1
   Compiling unic-common v0.9.0
   Compiling unic-char-range v0.9.0
   Compiling percent-encoding v2.1.0
   Compiling remove_dir_all v0.5.2
   Compiling byte-tools v0.3.1
   Compiling rayon-core v1.6.1
   Compiling same-file v1.0.5
   Compiling ordermap v0.3.5
   Compiling fixedbitset v0.1.9
   Compiling v_htmlescape v0.4.5
   Compiling xmlparser v0.10.0
   Compiling strum v0.15.0
   Compiling strsim v0.9.2
   Compiling void v1.0.2
   Compiling opaque-debug v0.2.3
   Compiling cc v1.0.50
   Compiling fake-simd v0.1.2
   Compiling bit-vec v0.5.1
   Compiling stringreader v0.1.1
   Compiling path-clean v0.1.0
   Compiling ascii v0.9.3
   Compiling diff v0.1.11
   Compiling crc32fast v1.2.0
   Compiling fixedbitset v0.2.0
   Compiling adler32 v1.0.4
   Compiling lalrpop-util v0.17.2
   Compiling num-derive v0.2.5
   Compiling jni-sys v0.3.0
   Compiling ucd-trie v0.1.2
   Compiling cesu8 v1.1.0
   Compiling lzw v0.10.0
   Compiling color_quant v1.0.1
   Compiling percent-encoding v1.0.1
   Compiling unicode-width v0.1.6
   Compiling scoped_threadpool v0.1.9
   Compiling base64 v0.11.0
   Compiling strsim v0.8.0
   Compiling vec_map v0.8.1
   Compiling ansi_term v0.11.0
   Compiling termcolor v0.3.6
   Compiling generational-arena v0.2.6
   Compiling thread_local v1.0.1
   Compiling thread_local v0.3.6
   Compiling semver v0.9.0
   Compiling rand_core v0.3.1
   Compiling rand_jitter v0.1.4
   Compiling phf_shared v0.7.24
   Compiling rand_chacha v0.1.1
   Compiling rand_pcg v0.1.2
   Compiling rand v0.6.5
   Compiling crossbeam-utils v0.7.0
   Compiling crossbeam-epoch v0.8.0
   Compiling indexmap v1.3.0
   Compiling num-traits v0.2.10
   Compiling num-integer v0.1.41
   Compiling num-rational v0.2.2
   Compiling num-iter v0.1.39
   Compiling c2-chacha v0.2.3
   Compiling itertools v0.8.2
   Compiling lock_api v0.3.2
   Compiling futures-channel v0.3.1
   Compiling futf v0.1.4
   Compiling nom v5.1.0
   Compiling nom v4.2.3
   Compiling error-chain v0.12.1
   Compiling owning_ref v0.3.3
   Compiling unicode-bidi v0.3.4
   Compiling unicode-normalization v0.1.11
   Compiling heck v0.3.1
   Compiling phf_shared v0.8.0
   Compiling unic-char-property v0.9.0
   Compiling unic-ucd-version v0.9.0
   Compiling block-padding v0.1.5
   Compiling walkdir v2.3.1
   Compiling petgraph v0.4.13
   Compiling unreachable v1.0.0
   Compiling roxmltree v0.7.3
   Compiling bit-set v0.5.1
   Compiling inflate v0.4.5
   Compiling gif v0.10.3
   Compiling textwrap v0.11.0
   Compiling rustc_version v0.2.3
   Compiling rand_hc v0.1.0
   Compiling rand_isaac v0.1.1
   Compiling rand_xorshift v0.1.1
   Compiling phf v0.7.24
   Compiling tendril v0.4.1
   Compiling unic-ucd-segment v0.9.0
   Compiling bibutils-sys v0.1.0 (/data/data/com.termux/files/home/texlab/texlab/crates/bibutils_sys)
   Compiling idna v0.2.0
   Compiling jni v0.11.0
error[E0425]: cannot find value `EXPECTED_JVM_FILENAME` in this scope
  --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/jni-0.11.0/build.rs:91:25
   |
91 |         if file_name == EXPECTED_JVM_FILENAME {
   |                         ^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
error: could not compile `jni`.
warning: build failed, waiting for other jobs to finish...
error: build failed
~~~

--- a/packages/texlab/build.sh
+++ b/packages/texlab/build.sh
@@ -1,0 +1,16 @@
+TERMUX_PKG_HOMEPAGE=https://texlab.netlify.app/
+TERMUX_PKG_DESCRIPTION="A cross-platform implementation of the Language Server Protocol providing rich cross-editing support for the LaTeX typesetting system"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=3.3.1
+TERMUX_PKG_SRCURL=https://github.com/latex-lsp/texlab.git
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_make() {
+	termux_setup_rust
+	cargo build --jobs $TERMUX_MAKE_PROCESSES --target $CARGO_TARGET_NAME --release
+}
+
+termux_step_make_install() {
+	install -Dm700 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/texlab
+}

